### DISCUSSION TITLE
Asynchronous HTTPS Server compatible with ESP32 series.

### DISCUSSION THREAD

**Discussion by xwtk** [espressif/arduino-esp32#8234] [General]
Hey everyone.

I have an ESP32-S3 which is currently running ESP32-HTTPS-Server library which hasn't been updated for a few years now, it is for communication between other ESP32s. It is very unstable in terms of SSL Handshake and often falls and even crashes the whole system even when 2 devices request simultaneously.

I followed the developers guide to make the system asynchronous, that is, to assign the function to a specific core using FreeRTOS. I even tried to assign all the other functions to one core and the server one to another, it still often terminates SSL Handshakes in the middle of the connection, and again, sometimes even crashes if 2 devices try to send a request simultaneously.

I have tried @me-no-dev 's ESPAsyncWebServer, and it works really well. Yet I fail to understand how a less-powerful ESP8266 is able to run an HTTPS server, while the ESP32 is not compatible with ESPAsyncWebServer's HTTPS mode.

The reason why I need to use an HTTPS communication is to ensure the information is encrypted asymmetrically on-the-fly with complete End-to-End. If you have any suggestions such as using a different protocol or a different library, or maybe you have an implementation of HTTPS Server based on ESPAsyncWebServer library, or even an Arduino-ESP32 compatible implementation of the ESP-IDF HTTPS Server library, I would very much appreciate if you could help.

For those wondering, I have tried ESP-NOW but it needs to have the keys stored already in the devices, and only supports up to 17 encrypted devices.

**Comment by HamzaHajeir** [comment]
I'm about to test TLS support inside [H4AsyncTCP](https://github.com/HamzaHajeir/H4AsyncTCP), in which an [H4AsyncWebServer](https://github.com/HamzaHajeir/H4AsyncWebServer) relies upon.

You might checkout current states of both. TLS isn't published yet. And it's paramount to use H4 Timing library and a customized [Arduino core build](https://github.com/HamzaHajeir/arduino-esp32/tree/lwip-tls).

**Comment by horendus** [comment]
Checkout https://github.com/khoih-prog/HTTPS_Server_Generic
